### ISSUE TITLE
Error: Failed to find a python interpreter in the .data section

### ISSUE THREAD

**Issue by anicecoffee** [benfred/py-spy#663]
I'm using the latest py-spy version 0.3.14 and I'm still getting this error for the following commands 

`py-spy top -- python SampleCode.py`

`py-spy record -o sample-profile.svg --format flamegraph -- python SampleCode.py`

I referred to the previous issue on this which was resolved but I couldn't troubleshoot for my system. I reverted to previous versions of py-spy but still ran into similar issues with the error now becoming 

`Error: Failed to find python version from target process`

Please help

**Comment by Archer-Code** [comment]
i have same question, I modified the source code and recompiled a new one. #662

**Comment by Gazalkhan** [comment]
I am experiencing the same issue and haven't found a solution yet. @benfred, could you please provide your input?

**Comment by tastyminerals** [comment]
Same issue, just tried running on OSX
```
sudo py-spy record -o profile.svg -- testme -from "60s"
Error: Failed to find python version from target process
```
trying with the PID also doesn't work :(
```
sudo py-spy record -o profile.svg --pid 88151
Password:
Error: Failed to find python version from target process
```

**Comment by benfred** [comment]
@anicecoffee @Gazalkhan I think this should be fixed by https://github.com/benfred/py-spy/pull/718 - can you test it out if you get a chance ? (you can download prebuilt wheels from the 'wheels' artifacts here https://github.com/benfred/py-spy/actions/runs/11619999861?pr=718#artifacts )

**Comment by benfred** [comment]
I think this should be fixed in the 0.4 release. Can you try it out and let me know?

If this is still not working please re-open the issue

**Comment by Archer-Code** [comment]
> I think this should be fixed in the 0.4 release. Can you try it out and let me know?
> 
> If this is still not working please re-open the issue

![image](https://github.com/user-attachments/assets/3261b0cf-a49d-4554-9945-3f615dd44d63)
